core/src/main/java/com/microsoft/applicationinsights/internal/schemav2/Envelope.java

```diff
@@ -45,7 +45,9 @@
     /**
      * Backing field for property Name.
      */
-    private String name;
+    private String namePart1;
+    private String namePart2;
+    private String namePart3;
```

Comment:
It took a minute to find what part 1, 2 and 3 are. Let's name them something like prefix, normalizedIkey, telemetryType or document in javadoc.

Reply:
👍